exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -98,6 +98,11 @@ type AzureManagedMachinePoolSpec struct {
 	// +kubebuilder:validation:Enum=Linux;Windows
 	// +optional
 	OSType *string `json:"osType,omitempty"`
+
+	// ScaleSetPriority specifies the ScaleSetPriority value. Default to Regular. Possible values include: 'Regular', 'Spot'
+	// +kubebuilder:validation:Enum=Regular;Spot
```

Comment:
@jackfrancis if you send the `Regular` value. AKS isn't setting it to `Regular` but returns the empty.

Later any other change to nodepool will fail as Azure API returns the error that it cannot change from `""` to `Regular`

Reply:
I wasn't able to reproduce this with labels changes to the node pool. Which node pool changes were you seeing this on when `ScaleSetPriority` was set to `"Regular"`?

Reply:
still doesn't work for me. 

steps to reproduce: 
1. create an agentpool with explicit `ScaleSetPriority` set to `Regular`
2. verify in `az aks nodepool show --cluster-name cluster-name -n nodepool-name` that it has `"scaleSetPriority": null,`
3. try to change a label in agentpool and see reconcile error `Message=\"Changing property 'properties.ScaleSetPriority' is not allowed.`

Reply:
Are you testing using a build from this branch? In this current PR implementation, I definitely can't reproduce. I do observe what you're seeing from the AKS API:

```
$ az aks nodepool show --cluster-name capz-e2e-l9d26h-aks -n pool1 -g capz-e2e-l9d26h-aks | grep 'scaleSetPriority'
  "scaleSetPriority": null,
```

However, I'm able to continually update the node pool using capz (using node labels below as an example):

```
$ k get azuremanagedmachinepool/pool1 -n capz-e2e-l9d26h -o yaml | grep -A 5 '  nodeLabels'
  nodeLabels:
    foo: bar
  osDiskSizeGB: 40
  osDiskType: Ephemeral
  osType: Linux
  providerIDList:
```

You can see I've already applied a `foo: bar` node label. I'll now add a 2nd label:

```
$ k edit azuremanagedmachinepool/pool1 -n capz-e2e-l9d26h
azuremanagedmachinepool.infrastructure.cluster.x-k8s.io/pool1 edited
$ k get azuremanagedmachinepool/pool1 -n capz-e2e-l9d26h -o yaml | grep -A 5 '  nodeLabels'
  nodeLabels:
    foo: bar
    hello: world
  osDiskSizeGB: 40
  osDiskType: Ephemeral
  osType: Linux
```

Now if I get the node pool data from the AKS API I see the updated label:

```
$ az aks nodepool show --cluster-name capz-e2e-l9d26h-aks -n pool1 -g capz-e2e-l9d26h-aks | grep -A 5 'nodeLabels'
  "nodeLabels": {
    "foo": "bar",
    "hello": "world"
  },
  "nodePublicIpPrefixId": null,
  "nodeTaints": [
```

As expected, `ScaleSetPriority` is unchanged from either source:

```
$ az aks nodepool show --cluster-name capz-e2e-l9d26h-aks -n pool1 -g capz-e2e-l9d26h-aks | grep 'scaleSetPriority'
  "scaleSetPriority": null,
$ k get azuremanagedmachinepool/pool1 -n capz-e2e-l9d26h -o yaml | grep '  scaleSetPriority'
  scaleSetPriority: Regular
```

I think this PR has overcome this particular pathology. We should ship add'l E2E tests with this PR to prove it, but I'm confident we're good here.

Reply:
definitely its working now, does Azure API changed something? 
in any case I agree to merge this